### PR TITLE
add ability to name formulas separately

### PR DESCRIPTION
**Pull request by CaseyFaist**
A tiny change that still needs to be tested that allows you to pass flag `--name` in to specify a name separately from the formula name

### CODE REVIEW

**Comment by dzuelke** [comment]
How does the name get exposed to the actual formula?

**Comment by CaseyFaist** [comment]
@dzuelke I'm not sure what you mean - the `name` arg is being set in the` __init__ `method of Formula [here](https://github.com/heroku-python/bob-builder/pull/37/files#diff-fc0dafe1f7669e28226034a1e14fe38fR39), which is how the command argument value gets passed through all the class methods

But `formula` just means the name of the script you're running in this project. This change is built with a default formula in mind that treats the tarball and Python version as variables that will be accessible from the environment, eg my experiment with a default template for python [here](https://github.com/heroku/nodebin/pull/54/files#diff-d796a88f4b1c3786610ba7834ef58dd4)

I wanted to change as little of bob for this as possible because I didn't want side effects for PHP builds - thus an optional commandline parameter. Docopt is not my fave tho, and I haven't tested for side effects thoroughly. Don't merge til I do plz :3 <3

File: bob/models.py [hunk 1]
```diff
@@ -155,15 +156,21 @@ def archive(self):
         print('Created: {}'.format(archive))
         self.archived_path = archive
 
-    def deploy(self, allow_overwrite=False):
+    def deploy(self, allow_overwrite=False, allow_name=None):
```
Comment (dzuelke) [inline bob/models.py]: This should not be "`allow_name`", but probably "`overridden_name`" or something.

File: bob/cli.py [hunk 2]
```diff
@@ -21,9 +22,10 @@
 from .utils import print_stderr
 
 
-def build(formula):
-    f = Formula(path=formula)
+def build(formula, name):
+    f = Formula(path=formula, name=name)
 
+    # print(formula)
```
Comment (dzuelke) [inline bob/cli.py]: Leftover test/debug `print`

File: bob/cli.py [hunk 3]
```diff
@@ -53,12 +55,14 @@ def main():
...
     do_deploy = args['deploy']
     do_overwrite = args['--overwrite']
+    do_name = args['--name']
 
     if do_build:
         build(formula)
 
     if do_deploy:
-        deploy(formula, overwrite=do_overwrite)
+        print(do_name)
```
Comment (dzuelke) [inline bob/cli.py]: Leftover test/debug `print`

**Comment by dzuelke** [comment]
What I meant was, let's say you have a base `python` formula, and do `bob deploy --name python-3.7.1 python`, then without somehow passing the name (`python-3.7.1`) to the formula, the build won't succeed, right?

Using an env var in the formula that holds the version is of course a workaround, but that env var needs to be exported for the `bob deploy` call, **and** be passed as `--name`, otherwise the build would fail (if you forget the env var) or write to the "generic" file name (if you forget the `--name` flag).

How about passing the name as the second argument to the formula?

**Comment by CaseyFaist** [comment]
that's a good idea! I'll still have to have bob pass the version as an argument to the script or set it as an env var for me based on how it's set up right now, but I like avoiding the workaround

File: bob/models.py (hunk 1 above)
Comment (CaseyFaist) [reply bob/models.py]: 👍

File: bob/cli.py (hunk 2 above)
Comment (CaseyFaist) [reply bob/cli.py]: 👍

File: bob/cli.py (hunk 3 above)
Comment (CaseyFaist) [reply bob/cli.py]: 👍

**Review by dzuelke — APPROVED** [review]
🚀

**Pull request merged**